src/sentry/static/sentry/app/actionCreators/projects.tsx

```diff
@@ -66,51 +76,53 @@ const _queryForStats = (api, projects, orgId) => {
   });
 };
 
-export const _debouncedLoadStats = debounce((api, projectSet, params) => {
-  const existingProjectStats = Object.values(ProjectsStatsStore.getAll()).map(
-    ({id}) => id
-  );
-  const projects = Array.from(projectSet).filter(
-    project => !existingProjectStats.includes(project)
-  );
+export const _debouncedLoadStats = debounce(
+  (api: Client, projectSet: Set<string>, params) => {
+    const storedProjects: {[key: string]: Project} = ProjectsStatsStore.getAll();
+    const existingProjectStats = Object.values(storedProjects).map(({id}) => id);
+    const projects = Array.from(projectSet).filter(
+      project => !existingProjectStats.includes(project)
+    );
 
-  if (!projects.length) {
-    _projectStatsToFetch.clear();
-    return;
-  }
-
-  // Split projects into more manageable chunks to query, otherwise we can
-  // potentially face server timeouts
-  const queries = chunk(projects, MAX_PROJECTS_TO_FETCH).map(chunkedProjects =>
-    _queryForStats(api, chunkedProjects, params.orgId)
-  );
-
-  Promise.all(queries)
-    .then(results => {
-      ProjectActions.loadStatsForProjectSuccess(
-        results.reduce((acc, result) => acc.concat(result), [])
-      );
-    })
-    .catch(() => {
-      addErrorMessage(t('Unable to fetch all project stats'));
-    });
+    if (!projects.length) {
+      _projectStatsToFetch.clear();
+      return;
+    }
 
-  // Reset projects list
-  _projectStatsToFetch.clear();
-}, 50);
+    // Split projects into more manageable chunks to query, otherwise we can
+    // potentially face server timeouts
+    const queries = chunk(projects, MAX_PROJECTS_TO_FETCH).map(chunkedProjects =>
+      _queryForStats(api, chunkedProjects, params.orgId)
+    );
 
-export function loadStatsForProject(api, project, params) {
+    Promise.all(queries)
```

Comment:
Could convert to `await if you wanted`

```ts
try {
   const results = await Promise.all(queries);
   ProjectActions.loadStatsForProjectSuccess(
     results.reduce((acc, result) => acc.concat(result), [])
   );
} catch {
  addErrorMessage(t('Unable to fetch all project stats'));
}

// Reset projects list
_projectStatsToFetch.clear();
```

Reply:
I'll make another pass to add async/await.